src/main/kotlin/org/jetbrains/research/ml/coding/assistant/dataset/model/DatasetRecord.kt

```diff
@@ -0,0 +1,77 @@
+package org.jetbrains.research.ml.coding.assistant.dataset.model
+
+import kotlinx.serialization.Serializable
+
+typealias Record = Map<String, String>
+
+/**
+ *  Model represents a single record entry in dataset
+ *  Model for a row in .csv file
+ */
+data class DatasetRecord(
+    val id: String,
+    val fragment: String,
+    val metaInfo: MetaInfo
+) {
+
+    internal constructor(record: Record) : this(
+        id = record[Keys.ID]!!,
+        fragment = record[Keys.FRAGMENT] ?: "",
+        metaInfo = MetaInfo(record)
+    )
+}
+
+/**
+ *  Model represents meta information attached to the dataset record
+ */
+@Serializable
+data class MetaInfo(
+    val age: Float?,
+    val programExperience: ProgramExperience?,
+    val testsResults: Double,
+    val task: DatasetTask
+) {
+    internal constructor(record: Record) : this(
+        age = record[Keys.AGE]?.toFloatOrNullWithDefault(-1.0f),
+        programExperience = record[Keys.PROGRAM_EXPERIENCE].toProgramExperience(),
+        testsResults = record[Keys.TESTS_RESULTS]!!.toDouble(),
+        task = DatasetTask.createFromString(record[Keys.TASK]!!)
+    )
+
+    val isFinalSolution: Boolean get() = testsResults == 1.0
+
+    enum class ProgramExperience {
+        LESS_THAN_HALF_YEAR,
+        FROM_HALF_TO_ONE_YEAR,
+        FROM_ONE_TO_TWO_YEARS,
+        FROM_TWO_TO_FOUR_YEARS,
+        FROM_FOUR_TO_SIX_YEARS,
+        MORE_THAN_SIX
+    }
+}
+
+private fun String?.toFloatOrNullWithDefault(default: Float): Float? {
+    val value = this?.toFloatOrNull()
+    if (value == default) {
+        return null
+    }
+    return value
+}
+
+private fun String?.toProgramExperience(): MetaInfo.ProgramExperience? =
+    this?.let {
+        try {
+            MetaInfo.ProgramExperience.valueOf(it)
+        } catch (e: IllegalArgumentException) {
+            null
+        }
+    }
+
+private object Keys {
```

Comment:
Maybe it would be better to use Enum class? Also, I think the name `Column` is better. What do you think?
```
enum class Column(val key: String) {
    FRAGMENT("fragment")
    ...
}
```

Reply:
what's the difference?

Reply:
You can use additional features Enum goes together with such as iterating over all keys at once or more simple usage of  `when`-block

Reply:
Don't use this feature here

Reply:
for now, you don't, right, but who knows